src/EnergyPlus/HybridEvapCoolingModel.cc

```diff
@@ -0,0 +1,1591 @@
+		{
+			double PLLatetRatio, PLVentRatio, PLSensibleRatio, PartRuntimeFraction;
+			PLLatetRatio = PLVentRatio = PLSensibleRatio = 0;
+
+			if (Mvent > 0)   PLVentRatio = MinOA_Msa / Mvent;
+			PartRuntimeFraction = 0; //PLVentRatio; tempory fix to deal with fact we haven't yet implimented a ventilation only backup mode.
+			if (SensibleRoomORZone != 0)   PLSensibleRatio = abs(RequestedConditioningLoad) / abs(SensibleRoomORZone);
+			if (PLSensibleRatio > PartRuntimeFraction) PartRuntimeFraction = PLSensibleRatio;
+			if (ZoneDehumidificationLoad > 0)  PLLatetRatio = RequestedLatentRoomORZone / ZoneDehumidificationLoad;
+			if (PLLatetRatio > PartRuntimeFraction) PartRuntimeFraction = PLLatetRatio;
+
+			return PartRuntimeFraction;
+		}
+
+		int Model::SetOperatingSetting(CStepInputs StepIns)
+		{
+
+
+			double Requestedlatent = 0; //needs to come from EP
+			bool DidWeMeetLoad = false;
+			bool DidWeMeetHumidificaiton = false;
+			bool DidWePartlyMeetLoad = false;
+			using General::RoundSigDigits;
+			int modenumber = 0;
+			int point_number = 0;
+			double MsaRatio = 0;
+			double OSAF = 0;
+			double Msa = 0;
+			double Mvent = 0;
+			double pOptimal_RunFractionTotalFuel = IMPLAUSIBLE_POWER;
+			double EIR, ElectricalPower, SHR, Tma, Wma, Hsa, Hma, TotalSystem;
+			//mode_optimal_power = 0;
+			double PreviousMaxiumConditioningOutput = 0;
+			double PreviousMaxiumHumidOrDehumidOutput = 0;
+			
+			
+			if (StepIns.RHosa > 1) { ShowSevereError("Unitary hybrid system error, required RH 0-1"); } //should be fractional 
+			if (StepIns.RHra > 1) { ShowSevereError("Unitary hybrid system error, required RH 0-1"); } //should be fractional 
+
+			double Wosa = CalcHum_ratio_W(StepIns.Tosa, StepIns.RHosa, 101.325);
+			double Wra = CalcHum_ratio_W(StepIns.Tra, StepIns.RHra, 101.325);
+			bool EnvironmentConditionsMet, EnvironmentConditionsMetOnce, MinVRMet, MinVRMetOnce, SAT_OC_Met, SAT_OC_MetOnce, SARH_OC_Met, SAHR_OC_MetOnce;
+			EnvironmentConditionsMetOnce = SAT_OC_Met = SAT_OC_MetOnce = SARH_OC_Met = SAHR_OC_MetOnce = false;
+
+			MinOA_Msa = StepIns.MinimumOA; //?why you copy it before?
+			//RequestedConditioningLoad = -StepIns.RequestedCoolingLoad / 1000; //convert to kw Cooling possitive now, heating negitive
+			
+			bool DehumidificationRequested = false;
+			bool HumidificationRequested = false;
+			// Load required to meet dehumidifying setpoint (<0 = a dehumidify load)  [kgWater/s]
+			if (StepIns.ZoneDehumidificationLoad < 0) DehumidificationRequested = true;
+			// Load required to meet humidifying setpoint (>0 = a humidify load) [kgWater/s]
+			if (StepIns.ZoneMoistureLoad > 0) HumidificationRequested = true;
+			//RequestedHumdification = StepIns.ZoneMoistureLoad;
+			//RequestedDeHumdification = StepIns.ZoneDehumidificationLoad;
+			double averaged_requestedLoad = 0;
+
+			std::list<CMode*>::const_iterator iterator;
```

Comment:
This is another chunk that adds to the EnergyPlus maintenance burden.  There should be much easier ways to do this than to create an iterator for a list of pointers, then get the first address, then increment it, then loop over the rest of the iterations.